.github/workflows/reusable-run-integration-tests.yml

```diff
@@ -0,0 +1,129 @@
+        required: false
+      fluentd_config_validator_api_key:
+        description: Fluentd config validator API Key
+        required: false
+      smtp_username:
+        description: SMTP username to test email notifications
+        required: false
+      smtp_password:
+        description: SMTP password to test email notifications
+        required: false
+      image_registry_username:
+        description: Password (if github use a PAT token) to use for authentication on private registry
+        required: false
+      image_registry_password:
+        description: Password (if github use a PAT token) to use for authentication on private registry
+        required: false
+      auth0_client_secret:
+        description: Auth0 client secret
+        required: false
+jobs:
+  call-run-integration-test-images-exist:
+    name: Test image exists
+    runs-on: ubuntu-latest
+    env:
+      IMAGE: ${{ inputs.cloud_image_registry }}/${{ inputs.cloud_image_name }}:${{ inputs.cloud_image_tag }}
+    steps:
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ inputs.cloud_image_registry }}
+          username: ${{ secrets.image_registry_username }}
+          password: ${{ secrets.image_registry_password }}
+      - run: docker pull $IMAGE
+
+  call-run-integration-tests:
+    name: run integration tests
+    needs: call-run-integration-test-images-exist
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          repository: ${{ inputs.target_repository }}
+          ref: ${{ inputs.target_repository_ref }}
+          token: ${{ secrets.image_registry_password }}
+          fetch-depth: 0
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ inputs.cloud_image_registry }}
+          username: ${{ secrets.image_registry_username }}
+          password: ${{ secrets.image_registry_password }}
+
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ inputs.go_version }}
+
+      - name: Create k8s Kind Cluster
+        if: ${{ inputs.enable_kind == true }}
+        uses: helm/kind-action@v1.2.0
+
+      - name: Go tests
+        run: |
+          go test ${{ env.GO_TEST_PARAMS }}
+        shell: bash
+        env:
+          GO_TEST_PARAMS: ${{ inputs.go_test_params }}
+          TEST_CLOUD_IMAGE: ${{ inputs.cloud_image_registry }}/${{ inputs.cloud_image_name }}
+          TEST_CLOUD_IMAGE_TAG: ${{ inputs.cloud_image_tag }}
+          TEST_FLUENTBIT_CONFIG_VALIDATOR_API_KEY: ${{ secrets.fluentbit_config_validator_api_key }}
+          TEST_FLUENTD_CONFIG_VALIDATOR_API_KEY: ${{ secrets.fluentd_config_validator_api_key }}
+          TEST_SMTP_USERNAME: ${{ secrets.smtp_username }}
+          TEST_SMTP_PASSWORD: ${{ secrets.smtp_password }}
+          AGGREGATOR_FLUENTBIT_TLS: off
+          AGGREGATOR_FLUENTBIT_TLS_VERIFY: off
```

Comment:
I think we'll want to verify this as well at some point so maybe expose with an off default input?

Reply:
Right now natively the listener at the API side doesn't have TLS support as that's driven by
cloudflare -> cloud-run when deployed, but for local development/testing we don't have TLS enabled in the cloud binary.